Running error on examples/pytorch/gat/train.py

## 🐛 Bug

When running the cpu version of gat, an error is reported, indicating that cuda is not supported.

## To Reproduce

Steps to reproduce the behavior:

python train.py --dataset=cora --gpu=-1

Traceback (most recent call last):
  File "train.py", line 181, in <module>
    main(args)
  File "train.py", line 110, in main
    torch.cuda.synchronize()
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dgl/lib/python3.8/site-packages/torch/cuda/__init__.py", line 493, in synchronize
    _lazy_init()
  File "/opt/homebrew/Caskroom/miniconda/base/envs/dgl/lib/python3.8/site-packages/torch/cuda/__init__.py", line 210, in _lazy_init
    raise AssertionError("Torch not compiled with CUDA enabled")
AssertionError: Torch not compiled with CUDA enabled


## Expected behavior

The example should run successful.

## Environment

 - DGL Version (e.g., 1.0): 0.8.1
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3):1.11.0
 - OS (e.g., Linux): macOS 12.3.1 
 - How you installed DGL (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source): 
 - Python version: 3.8.13
 - CUDA/cuDNN version (if applicable):
 - GPU models and configuration (e.g. V100):
 - Any other relevant information:

## Additional context

<!-- Add any other context about the problem here. -->

It seems that the conditional judgment of cuda is missing here (line 110 and 121).

https://github.com/dmlc/dgl/blob/4b87e47f15552cb27ddf8e594769ad0361d06627/examples/pytorch/gat/train.py#L109-L111

https://github.com/dmlc/dgl/blob/4b87e47f15552cb27ddf8e594769ad0361d06627/examples/pytorch/gat/train.py#L120-L122

--- a/examples/pytorch/gat/train.py
+++ b/examples/pytorch/gat/train.py
@@ -107,7 +107,8 @@ def main(args):
     for epoch in range(args.epochs):
         model.train()
         if epoch >= 3:
-            torch.cuda.synchronize()
+            if cuda:
+                torch.cuda.synchronize()
             t0 = time.time()
         # forward
         logits = model(features)
@@ -118,7 +119,8 @@ def main(args):
         optimizer.step()
 
         if epoch >= 3:
-            torch.cuda.synchronize()
+            if cuda:
+                torch.cuda.synchronize()
             dur.append(time.time() - t0)
 
         train_acc = accuracy(logits[train_mask], labels[train_mask])

